test/test_support.ts

```diff
@@ -40,25 +40,31 @@ export const compilerOptions: ts.CompilerOptions = {
 };
 
 const {cachedLibPath, cachedLib} = (() => {
-  const host = ts.createCompilerHost(compilerOptions);
-  const fn = host.getDefaultLibFileName(compilerOptions);
-  const p = ts.getDefaultLibFilePath(compilerOptions);
+  const host = ts.createCompilerHost(goldenCompilerOptions);
+  const fn = host.getDefaultLibFileName(goldenCompilerOptions);
+  const p = ts.getDefaultLibFilePath(goldenCompilerOptions);
   return {
     // Normalize path to fix mixed/wrong directory separators on Windows.
     cachedLibPath: path.normalize(p),
-    cachedLib: host.getSourceFile(fn, ts.ScriptTarget.ES2015)
+    cachedLib: host.getSourceFile(fn, goldenCompilerOptions.target!),
   };
 })();
 
 /** Creates a ts.Program from a set of input files. */
 export function createProgram(
     sources: Map<string, string>,
-    tsCompilerOptions: ts.CompilerOptions = compilerOptions): ts.Program {
+    tsCompilerOptions: ts.CompilerOptions = goldenCompilerOptions): ts.Program {
   return createProgramAndHost(sources, tsCompilerOptions).program;
 }
 
+export function createHost(
```

Comment:
createHost calling createProgramAndHost and taking sources seems surprising. Maybe we should re-structure the call sequence to extract a caching `createHost`?